docs/development.md

```diff
@@ -150,6 +150,36 @@ Our Concourse pipeline definitions are kept in the [cf-operator-ci](https://gith
 - create functions in `env/machine`
 - create functions in `env/catalog`
 
+## Create-Or-Update pattern
+
+A pattern that comes up quite often is that an object needs to be updated if it already exists or created if it doesn't. `controller-runtime` provides the `controller-util` package which has a `CreateOrUpdate` function that can help with that. It takes a skeleton object and a reconcile function that is caleld in both the create and the update case:
+
+```go
+tempManifestSecret := &corev1.Secret{
+  ObjectMeta: metav1.ObjectMeta{
+    Name:      tempManifestSecretName,
+    Namespace: instance.GetNamespace(),
+  },
+}
+_, err = controllerutil.CreateOrUpdate(ctx, r.client, tempManifestSecret, func(obj runtime.Object) error {
+  s := obj.(*corev1.Secret)
```

Comment:
The example needs to avoid `panic` - https://tour.golang.org/methods/15
Please use  something like:
```
s, ok := obj.(*corev1.Secret)
if !ok 
{ 
  return errors.New("object is not a secret") 
}
```